### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by rastaman111** [LeoNatan/LNPopupController#525] [General] [answered]
Hi @LeoNatan 

1. How can I round the top edges of a view?
2. Many users have noticed that it is difficult to lower the view down and you should try to do it

Example: in Apple music view falls easily

![IMG_2658](https://user-images.githubusercontent.com/15727518/201300557-0cd6aad5-2aa0-42fa-9fba-8a4eb62d2fbb.PNG)

**Comment by LeoNatan** [comment]
Currently, presentation is full screen only. You can fake it, but won’t look as polished as the Apple implementation.

**Reply by rastaman111** [reply]
Yes, I know that in full screen mode.  I lowered the screen a little to show exactly what needs to be rounded.

**Reply by LeoNatan** [reply]
You can create a mask with a rounded top corner, and apply it to the popup content view, which will create a similar effect.

**Reply by LeoNatan** [reply, marked as answer]
https://developer.apple.com/documentation/uikit/uiview/1622557-maskview

**Reply by LeoNatan** [reply]
Also see
https://stackoverflow.com/questions/10167266/how-to-set-cornerradius-for-only-top-left-and-top-right-corner-of-a-uiview

**Reply by rastaman111** [reply]
@LeoNatan 

What about the heavy lowering of the controller, even in the example it is clear that the second time it turns out to lower it, if the thought is how to fix it?

**Reply by LeoNatan** [reply]
I’m sorry, I do t understand your question. The presentation is full screen. If you want to fake a “hole” at the top, you can do it using a mask view.

**Reply by LeoNatan** [reply]
There is nothing to “fix”. Implementing what you want is a big feature that is out of scope for the project currently. Apple Music these days just uses a custom sheet presentation controller (https://developer.apple.com/documentation/uikit/uisheetpresentationcontroller), with a lot less custom logic than it used to have (and this framework implemented years ago).
